swri_roscpp/CMakeLists.txt

```diff
@@ -88,7 +109,8 @@ if (BUILD_TESTING)
   )
 
   ament_add_gtest(topic_service_test_server test/topic_service_test.cpp)
-  rosidl_target_interfaces(topic_service_test_server ${PROJECT_NAME} "rosidl_typesupport_cpp")
+  rosidl_get_typesupport_target(cpp_typesupport_target "${PROJECT_NAME}" "rosidl_typesupport_cpp")
```

Comment:
This will fail to compile on ROS Galactic or earlier because `rosidl_get_typesupport_target` does not exist.

You could replace this with something like:
```cmake
if ("$ENV{ROS_DISTRO}" STRGREATER "galactic")
  rosidl_get_typesupport_target(cpp_typesupport_target "${PROJECT_NAME}_test" "rosidl_typesupport_cpp")
  target_link_libraries(topic_service_test_server "${cpp_typesupport_target}")
else()
  rosidl_target_interfaces(topic_service_test_server "${PROJECT_NAME}_test" "rosidl_typesupport_cpp")
endif()
```

Note that `rosidl_generate_interfaces(${PROJECT_NAME}` above would also need to be changed to `rosidl_generate_interfaces(${PROJECT_NAME}_test` to reflect the new target name.

Reply:
Can also confirm it's broken on prior ROS2 releases.